pkg/ipam/types.go

```diff
@@ -114,3 +115,17 @@ func (ipam *IPAM) DebugStatus() string {
 	ipam.allocatorMutex.RUnlock()
 	return str
 }
+
+// GetVpcCIDRs returns all the CIDRs associated with the VPC this node belongs to.
+// This works only cloud provider IPAM modes and returns nil for other modes.
+func (ipam *IPAM) GetVpcCIDRs() (vpcCIDRs []*cidr.CIDR) {
```

Comment:
An alternative approach to exposing this would be to have the VPC CIDR pushed to the `nodediscovery.NodeDiscovery` object which is increasingly becoming the source of truth for anything related to the local node. I assume it might eventually replace the global getters in the `node` package (which is currently used by `restoreCiliumHostIPs` to get possible CIDRs in other IPAM mode).

The IPAM module already uses it to register additional IP allocation CIDRs here:
https://github.com/cilium/cilium/blob/58e1f35d3317b1baed82db7fb12e66e7d845d5a4/pkg/ipam/clusterpool.go#L504-L505

We could do something similar with the VPC CIDRs. But I think the current solution also works.

Reply:
It does not look straightforward to include this in NodeDiscovery. Looked into reusing the existing pod CIDR info on NodeDiscovery struct, but that's not initialized for ENI mode before the router IP restoration happens. Unless there are other objections, can i continue with the current approach ?